metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -378,34 +397,39 @@ void build_pred_succ(const std::string &graph_fname,
         std::vector<bool> pred_boundary_buf;
 
         for (uint64_t i = start; i < std::min(start + BS, graph.num_nodes() + 1); ++i) {
-            BOSS::edge_index boss_idx = graph.kmer_to_boss_index(i);
-            if (!dummy[boss_idx]) {
-                BOSS::edge_index next = boss.fwd(boss_idx);
-                assert(next);
-                if (!dummy[next]) {
-                    while (rd_succ.size() && !rd_succ[next]) {
-                        next--;
-                        assert(!boss.get_last(next));
-                    }
-                    succ_buf.push_back(to_row(graph.boss_to_kmer_index(next)));
+            bool skip_succ = false, skip_all = false;
+            if (succinct) { // Legacy code for DBGSuccinct
+                BOSS::edge_index boss_idx = from_graph_index(graph, i);
+                if((*dummy)[boss_idx]) {
+                    skip_all = true;
```

Comment:
I added this check, and also a check below that a predecessor is not dummy to fix integration tests. But still, I feel like it shouldn't ever happen, unless I misunderstand the concept of BOSS dummy edges.